src/libraries/xchaininit/src/xinit.cpp

```diff
@@ -418,6 +431,12 @@ int topchain_noparams_init(const std::string& pub_key, const std::string& pri_ke
     xinfo("=== xtopchain start here with noparams ===");
     std::cout << "xnode start begin..." << std::endl;
 
+    //init data_path into xvchain instance
+    base::xvchain_t::instance().set_data_dir_path(datadir);
+    //init auto_prune feature
+    set_auto_prune_switch(XGET_CONFIG(auto_prune_data));
```

Comment:
this API set_auto_prune_switch(...) is called in two places because we have xtopchain & topio, thus call it in xapplication_t::start() is a better choice? we don't care if it's running under xtopchain mode or topio mode.